triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala

```diff
@@ -0,0 +1,371 @@
+
+  def loadFromConfigFile: Option[Either[ConfigReaderFailures, TriggerServiceAppConf]] =
+    configFile.map(cf => ConfigSource.file(cf).load[TriggerServiceAppConf])
+
+  def loadFromCliArgs: ServiceConfig = {
+    ServiceConfig(
+      darPaths = darPaths,
+      address = address,
+      httpPort = httpPort,
+      ledgerHost = ledgerHost,
+      ledgerPort = ledgerPort,
+      authInternalUri = authInternalUri,
+      authExternalUri = authExternalUri,
+      authBothUri = authBothUri,
+      authRedirectToLogin = authRedirectToLogin,
+      authCallbackUri = authCallbackUri,
+      maxInboundMessageSize = maxInboundMessageSize,
+      minRestartInterval = minRestartInterval,
+      maxRestartInterval = maxRestartInterval,
+      maxAuthCallbacks = maxAuthCallbacks,
+      authCallbackTimeout = authCallbackTimeout,
+      maxHttpEntityUploadSize = maxHttpEntityUploadSize,
+      httpEntityUploadTimeout = httpEntityUploadTimeout,
+      timeProviderType = timeProviderType,
+      commandTtl = commandTtl,
+      init = init,
+      jdbcConfig = jdbcConfig,
+      portFile = portFile,
+      allowExistingSchema = allowExistingSchema,
+      compilerConfig = compilerConfig,
+    )
+  }
+
+  def loadConfig: Option[ServiceConfig] =
+    loadFromConfigFile.cata(
+      {
+        case Right(cfg) => Some(cfg.toServiceConfig)
+        case Left(ex) =>
+          logger.error(
+            s"Error loading trigger service config from file ${configFile}",
+            ex.prettyPrint(),
+          )
+          None
+      },
+      Some(loadFromCliArgs),
+    )
+}
+
+private[trigger] object Cli {
+
+  val DefaultHttpPort: Int = 8088
+  val DefaultMaxInboundMessageSize: Int = RunnerConfig.DefaultMaxInboundMessageSize
+  val DefaultMinRestartInterval: FiniteDuration = FiniteDuration(5, duration.SECONDS)
+  val DefaultMaxRestartInterval: FiniteDuration = FiniteDuration(60, duration.SECONDS)
+  // Adds up to ~1GB with DefaultMaxInboundMessagesSize
+  val DefaultMaxAuthCallbacks: Int = 250
+  val DefaultAuthCallbackTimeout: FiniteDuration = FiniteDuration(1, duration.MINUTES)
+  val DefaultMaxHttpEntityUploadSize: Long = RunnerConfig.DefaultMaxInboundMessageSize.toLong
+  val DefaultHttpEntityUploadTimeout: FiniteDuration = FiniteDuration(1, duration.MINUTES)
+  val DefaultCompilerConfig: Compiler.Config = Compiler.Config.Default
+  val DefaultCommandTtl: FiniteDuration = FiniteDuration(30, duration.SECONDS)
+
+  private[trigger] def redirectToLogin(value: String): AuthClient.RedirectToLogin = {
+    value.toLowerCase match {
+      case "yes" => AuthClient.RedirectToLogin.Yes
+      case "no" => AuthClient.RedirectToLogin.No
+      case "auto" => AuthClient.RedirectToLogin.Auto
+      case s =>
+        throw new IllegalArgumentException(s"value '$s' is not one of 'yes', 'no', or 'auto'.")
+    }
+  }
+
+  implicit val redirectToLoginRead: scopt.Read[AuthClient.RedirectToLogin] =
+    scopt.Read.reads(redirectToLogin)
+
+  private[trigger] val Empty = Cli(
```

Comment:
```suggestion
  private[trigger] val Default = Cli(
```
Doesn't look particularly empty (don't accept this suggestion as is, I assume it'll break plenty of stuff).

Reply:
Will change it to `Default` 👍